library/src/main/java/com/novoda/downloadmanager/MigrationJob.java

```diff
@@ -41,35 +49,51 @@ public void run() {
         MigrationExtractor migrationExtractor = new MigrationExtractor(database, filePersistence);
+        List<Migration> partialMigrations = partialDownloadMigrationExtractor.extractMigrations();
+        List<Migration> completeMigrations = migrationExtractor.extractMigrations();
         DownloadsPersistence downloadsPersistence = RoomDownloadsPersistence.newInstance(context);
         LocalFilesDirectory localFilesDirectory = new AndroidLocalFilesDirectory(context);
         UnlinkedDataRemover unlinkedDataRemover = new UnlinkedDataRemover(downloadsPersistence, localFilesDirectory);
 
+        int totalNumberOfMigrations = partialMigrations.size() + completeMigrations.size();
+        InternalMigrationStatus migrationStatus = new VersionOneToVersionTwoMigrationStatus(
+                jobIdentifier,
+                MigrationStatus.Status.DB_NOT_PRESENT,
+                0,
+                totalNumberOfMigrations
+        );
+
         unlinkedDataRemover.remove();
         migrationStatus.markAsExtracting();
         onUpdate(migrationStatus);
 
-        Log.d(TAG, "about to extract migrations, time is " + System.nanoTime());
+        migrationStatus.markAsMigrating();
+        onUpdate(migrationStatus);
 
-        String basePath = filePersistence.basePath().path();
-        migratePartialDownloads(database, partialDownloadMigrationExtractor, downloadsPersistence, basePath);
-        migrateCompleteDownloads(migrationStatus, database, migrationExtractor, downloadsPersistence, basePath);
+        migratePartialDownloads(migrationStatus, database, partialMigrations, downloadsPersistence, basePath);
+        migrateCompleteDownloads(migrationStatus, database, completeMigrations, downloadsPersistence, basePath);
+        deleteVersionOneDatabase(migrationStatus, database);
+
+        migrationStatus.markAsComplete();
+        onUpdate(migrationStatus);
     }
 
-    private void onUpdate(MigrationStatus migrationStatus) {
+    private void onUpdate(InternalMigrationStatus migrationStatus) {
         for (MigrationCallback migrationCallback : migrationCallbacks) {
-            migrationCallback.onUpdate(migrationStatus);
+            migrationCallback.onUpdate(migrationStatus.copy());
         }
     }
 
-    private void migratePartialDownloads(SqlDatabaseWrapper database,
-                                         PartialDownloadMigrationExtractor partialDownloadMigrationExtractor,
+    private void migratePartialDownloads(InternalMigrationStatus migrationStatus,
+                                         SqlDatabaseWrapper database,
+                                         List<Migration> partialMigrations,
                                          DownloadsPersistence downloadsPersistence,
                                          String basePath) {
-        List<Migration> partialMigrations = partialDownloadMigrationExtractor.extractMigrations();
+        Log.d(TAG, "Partial migrations starting " + System.nanoTime());
```

Comment:
Removed a lot of the logs from this class so that we notify before and after a method executes.